src/cli/cli.cpp

```diff
@@ -708,6 +719,93 @@ void Interpreter::ProcessDiscover(int argc, char *argv[])
     AppendResult(error);
 }
 
+#if OPENTHREAD_ENABLE_DNS_CLIENT
+void Interpreter::ProcessDns(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+    long port = OT_DNS_DEFAULT_DNS_SERVER_PORT;
+    Ip6::MessageInfo messageInfo;
+    otDnsQuery query;
+
+    if (argc >= 2 && strcmp(argv[0], "resolve") == 0)
```

Comment:
To reduce the number of if statements, especially if we want to add more DNS arguments down the line, could the argc check be moved to a `VerifyOrExit`?
```
VerifyOrExit(argc >= 2, error = kThreadError_InvalidArgs);
```